components/renku_data_services/connected_services/models.py

```diff
@@ -0,0 +1,81 @@
+"""Models for connected services."""
+
+from dataclasses import dataclass
+from datetime import UTC, datetime
+from typing import Any
+
+from pydantic import BaseModel
+
+from renku_data_services.connected_services.apispec import ConnectionStatus, ProviderKind
+
+
+@dataclass(frozen=True, eq=True, kw_only=True)
+class OAuth2Client(BaseModel):
```

Comment:
This comment is quite confusing as this is the same approach used as for other models. Also, the code makes uses of `model_validate` from `pydantic.

Reply:
As discussed you dont really need these because you are using the pydantic classes generated from the apispec to generate the http response bodies. So you can remove these in a follow up pr.

Reply:
Done in a7c16362af57d2739014830b74f733aea0aeede5.